File: projects/inflection/modules/prod/prod_py/ru/_declension.py
from projects.inflection.modules.dev.dev_py import a
from projects.inflection.modules.dev.dev_py import mw
from projects.inflection.modules.dev.dev_py import tools as _

dev_prefix = 'User:Vitalik/'  # comment this on `prod` version


from .declension import _init as init
from .declension.run.result import error as e
from .declension import _run as run


module = 'declension'  # local


@a.starts(module)
def forms(func, base, args, frame):  # export  # todo: rename to `out_args`
    i = init.init_info(base, args, frame)  # local
    if e.has_error(i):
        return _.returns(module, func, i.result)
    # end

    # INFO: Запуск основного алгоритма и получение результирующих словоформ:
    run.run(i)

    return _.returns(module, func, i.result)
# end


# return export

File: projects/inflection/modules/prod/prod_lua/ru/_declension.lua
local dev_prefix = ''
dev_prefix = 'User:Vitalik/'  -- comment this on `prod` version

local export = {}
local _ = require('Module:' .. dev_prefix .. 'inflection/tools')


local init = require('Module:' .. dev_prefix .. 'inflection/ru/declension/init')  -- '_' /init
local e = require('Module:' .. dev_prefix .. 'inflection/ru/declension/run/result/error')  -- 'declension.'
local run = require('Module:' .. dev_prefix .. 'inflection/ru/declension/run')  -- '_' /run


local module = 'declension'


-- @starts
function export.forms(base, args, frame)  -- todo: rename to `out_args`
	func = "forms"
	_.starts(module, func)

	local i = init.init_info(base, args, frame)
	if e.has_error(i) then
		return _.returns(module, func, i.result)
	end

--	INFO: Запуск основного алгоритма и получение результирующих словоформ:
	run.run(i)

	return _.returns(module, func, i.result)
end


return export
